src/ophys_etl/modules/roi_cell_classifier/compute_classifier_artifacts.py

```diff
@@ -0,0 +1,275 @@
+        description="Path to pickle file containing full movie graph.",
+    )
+
+    # Output Artifact location.
+    out_dir = fields.OutputDir(
+        required=True,
+        description="Output directory to put artifacts.",
+    )
+
+    # Artifact generation settings.
+    low_quantile = fields.Float(
+        required=False,
+        defalut=0.2,
+        description="Low quartile to saturate/clip to.",
+    )
+    high_quantile = fields.Float(
+        required=False,
+        defalut=0.99,
+        description="High quartile to saturate/clip to.",
+    )
+    cutout_size = fields.Int(
+        required=False,
+        defalut=128,
+        description="Size of square cutout.",
+    )
+
+
+class ClassifierArtifactsGenerator(ArgSchemaParser):
+    """Create cutouts from the average, max, correlation projects for each
+    detected ROI. Additionally return the ROI mask. Store outputs as 8bit
+    pngs.
+    """
+
+    default_schema = ClassifierArtifactsInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+        self.logger.setLevel(self.args["log_level"])
+
+        ophys_etl_commit_sha = os.environ.get("OPHYS_ETL_COMMIT_SHA",
+                                              "local build")
+        self.logger.info(f"OPHYS_ETL_COMMIT_SHA: {ophys_etl_commit_sha}")
+        t0 = time.time()
+
+        out_dir = pathlib.Path(self.args["out_dir"])
+        if not out_dir.exists():
+            out_dir.mkdir()
+        assert out_dir.is_dir()
+
+        video_path = pathlib.Path(self.args["video_path"])
+        assert video_path.is_file()
+
+        exp_id = video_path.name.split("_")[0]
+
+        roi_path = pathlib.Path(self.args["roi_path"])
+        assert roi_path.is_file()
+
+        graph_path = pathlib.Path(self.args["graph_path"])
+        assert graph_path.is_file()
+
+        proj = self.get_max_avg(video_path)
+        self.logger.info("Calculated mean and max images...")
+        avg_img = proj["avg"]
+        max_img = proj["max"]
+        corr_img = graph_to_img(graph_path)
+        self.logger.info("Calculated correlation image...")
+
+        quantiles = (self.args["low_quantile"], self.args["high_quantile"])
+        q0, q1 = np.quantile(max_img, quantiles)
+        max_img = normalize_array(array=max_img,
+                                  lower_cutoff=q0,
+                                  upper_cutoff=q1)
+
+        q0, q1 = np.quantile(avg_img, quantiles)
+        avg_img = normalize_array(array=avg_img,
+                                  lower_cutoff=q0,
+                                  upper_cutoff=q1)
+
+        q0, q1 = np.quantile(corr_img, quantiles)
+        corr_img = normalize_array(array=corr_img,
+                                   lower_cutoff=q0,
+                                   upper_cutoff=q1)
+
+        self.logger.info("Normalized images...")
+
+        with open(roi_path, "rb") as in_file:
+            extract_roi_list = sanitize_extract_roi_list(
```

Comment:
this isn't a request for a change. Just a lament.

We still technically haven't defined the output schema of segmentation in production. Looks like we are just going to lock ourselves into what we've been using because it's what we have.... 

and *that* is how you build scientific software

Reply:
An output schema allows you to output a dict by calling `self.output` in an `argschema.ArgschemaParser` object. It basically is equivalent to `output_file.write(json.dumps(my_data))` with an extra layer of validation on top to make sure you match the schema (i.e. an exception will get thrown if you try to output something that doesn't conform to the output schema). See, for instance,

https://github.com/AllenInstitute/ophys_etl_pipelines/blob/main/src/ophys_etl/modules/suite2p_registration/__main__.py#L297

https://github.com/AllenInstitute/ophys_etl_pipelines/blob/main/src/ophys_etl/modules/suite2p_registration/__main__.py#L25

or

https://github.com/AllenInstitute/ophys_etl_pipelines/blob/main/src/ophys_etl/modules/suite2p_registration/__main__.py#L297

https://github.com/AllenInstitute/ophys_etl_pipelines/blob/main/src/ophys_etl/modules/suite2p_registration/__main__.py#L25

Usually we use it when the output.json needs to be read in by a downstream LIMS queue, or ingested into LIMS database columns.

Whenever we get around to transitioning Suite2P segmentation to production, we will need to specify that the ROIs get written out with a specific schema, since there will be downstream queues that need to

a) find the ROIs (I assume the ROI JSONs will be well known files; maybe they won't)
b) ingest the ROIs for classification

We've been working in prototype space so long, I have no idea how "productionization of Suite2P segmentation" will go.

Reply:
Okay, sounds like we'll cross that bridge when we come to it.

Reply:
Yes. Like I said: not a request for action; just a lament of the state of things.